### ISSUE TITLE
[Bug] E2E tests create a new test server for every batch of tests, sometimes failing to bind to port

### ISSUE THREAD

**Issue by bershanskiy** [darkreader/darkreader#6889]
# Bug Report

## Bug Description
If you run `npm run tests:chrome`, tests can occasionally fail with error "address already in use" in `createPopupTestServer()`. This happens because `createPopupTestServer()` is called for every test file separately and occasionally test port fails to become available before the next test starts. On local machine, this bug occurs rarely, but on CI it happens every time.

## Website Address

N/A

## Steps To Reproduce
Apply the following patch:
```diff
diff --git a/tests/browser/environment.js b/tests/browser/environment.js
index f20d64ba..edf4befa 100644
--- a/tests/browser/environment.js
+++ b/tests/browser/environment.js
@@ -166,6 +166,7 @@ class PuppeteerEnvironment extends JestNodeEnvironment {
     }

     async createPopupTestServer() {
+        console.log('CALLED CREATE TESTS SERVER')
         // Puppeteer cannot evaluate scripts in moz-extension:// pages
         // https://github.com/puppeteer/puppeteer/issues/6616
         return new Promise((resolve) => {
```
And run `npm run test:chrome`. Observe that `CALLED CREATE TESTS SERVER` is printed multiple times.

## Expected Behavior

Ideally, server should be created only once and then reused.

## Actual Behavior

Server is created multiple times.

## Screenshots

N/A

## System Information

- OS: Windows 10
- Browser: N/A
- Dark Reader Version: N/A

## Additional Context

Fixing this bug is necessary for running E2E tests on CI. We can fix it by creating a single "index" file for all E2E tests.

**Comment by Gusted** [comment]
Jest doesn't seem to support a `beforeAll` and `afterAll`(with puppeteer) and is really focussed into having each tests their "own" sandbox. We could make a simple helper function that ensure the Websocket server is different for each test suite.

**Comment by bershanskiy** [comment]
Do you mean we just bind to a new port number for every single test suite?

**Comment by Gusted** [comment]
Yeah, seems like the best option for now. If the error is about `address already in use` and turns out to be the popup then we simple should make a new port number on each test suite to avoid that.
